x-pack/plugins/cases/public/components/custom_fields/form.tsx

```diff
@@ -49,17 +50,27 @@ const formDeserializer = ({
 }: CustomFieldConfiguration): CustomFieldsConfigurationFormProps => {
   return {
     key,
+    options: { required: Boolean(required) },
     label,
     type,
-    options: { required: Boolean(required) },
   };
 };
 
-const FormComponent: React.FC<Props> = ({ onChange }) => {
-  const keyDefaultValue = useMemo(() => uuidv4(), []);
+const FormComponent: React.FC<Props> = ({ onChange, initialValue }) => {
+  const keyDefaultValue = useMemo<string>(() => {
```

Comment:
super nit: I think is fine if we keep it like `const keyDefaultValue = useMemo(() => uuidv4(), []);` for simplicity.